### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by nwakauc**
### Helloo CR ✋ , In this project I have managed to make the budget app, and did my best to implement the project requirements as follows 🧧;

- [ ] `Sign up` and `log in` pages;
- [ ] The `user` should be able to register in the app with `Name,` `email` and `password` (all mandatory).
- [ ] The `user` can log into the app using email and password.
- [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

### Home page (categories page)
- [ ] When the `user` logs in, they are presented with the `categories` page.
- [ ] For each category, the `user` can see their `name,` `icon` and the total amount of all the transactions that belongs to that category.
- [ ] When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page
- [ ] For a given `category,` the list of `transactions` is presented, ordered by the most recent.
- [ ] At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- [ ] There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- [ ] When the user clicks on the "Back" button `(<)`, the user navigates to the home page.
- [ ] Add a new category" page
- [ ] The `user` fills out a form to create a new category, indicating their name and icon (both mandatory).
- [ ] The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- [ ] When the user clicks on the "Back" button `(<)`, the user navigates to the home page.
### Add a new transaction page
- [ ] The `user` fills out a form to create a new transaction with:
`name` (mandatory)
`amount` (mandatory)
`categories` (mandatory at least one)
- [ ] The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- [ ] When the `user` clicks on the "Back" button `(<)`, the user navigates to the transactions page for that category.
- [ ] **Thank you for taking the time to review my code I look forward to your feedback👏**

### CODE REVIEW

File: app/views/categories/_expense_item.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        <div class="item-icon">💸</div>
+        <div class="item-group">
+            <h3 class="item-name"><%= expense.name %></h3>
+            <p class="item-data">
+                <%= expense.updated_at.to_datetime.strftime("%B %d, %Y") %>
+            </p>
+        </div>
+    </div>
+    <div class="item-group">
+        <p class="item-value">
```
Comment (Lucas-Masaba) [inline app/views/categories/_expense_item.html.erb:12]: - [ ] On the home page, the total amount spent on a given category is not shown. It only shows $0.00. Please make sure that the total amount spent on a particular category is shown on the home page :+1: 
image

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<%= render 'shared/header', title: "Categories", back_link: false %>
+<section class="container">
+<div class="total-header">
+    <% expense_count = @unique_expenses.size %>
+    <p>Spending: <%= number_to_currency(@total_expense) %></p>
```
Comment (Lucas-Masaba) [inline app/views/categories/index.html.erb:5]: [OPTIONAL]
- The total amount of spending you added to the home page does not change when a new expense is added. 
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @uchexm  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks are passing :heavy_check_mark: 
- Well-written pull request message :writing_hand: 
- Good use of gitflow :+1: 
- Your UI design looks really good. Probably the best I've seen for this project in my opinion :star_struck: 
- All written tests are passing :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/_expense_item.html.erb (hunk 1 above)
Comment (uchexm) [reply app/views/categories/_expense_item.html.erb:12]: On it ✌

Comment (uchexm) [reply app/views/categories/_expense_item.html.erb:12]: On it ✌

File: app/views/categories/index.html.erb (hunk 2 above)
Comment (uchexm) [reply app/views/categories/index.html.erb]: Done and done 😎

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @uchexm ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**